### PR TITLE
RetinaNet Ragged image Issue

### LINKED ISSUE #1359: Raise a helpful error message from OD models when receiving a Ragged image input
Currently the error message is very arcane.  We can remedy this and greatly boost the UX.

We can instruct users to use:

- JitteredResize()
- Resizing(pad_to_aspect_ratio=True)

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?

fixes #1359

@LukeWood @ianstenbit

### CODE REVIEW

File: keras_cv/models/object_detection/retina_net/retina_net.py [hunk 1]
```diff
@@ -218,6 +218,11 @@ def _forward(self, images, training=None):
         return box_pred, cls_pred
 
     def call(self, images, training=None):
+        if isinstance(images, tf.RaggedTensor):
```
Comment (bhack) [inline keras_cv/models/object_detection/retina_net/retina_net.py:221]: Can we add in the test an expected failed test for the ragged input? You could use the same decorator as in:
https://github.com/keras-team/keras-cv/blob/32701437b713050de951351529335f1fc725ceb0/keras_cv/layers/object_detection/multi_class_non_max_suppression_test.py#L50

Comment (IMvision12) [reply keras_cv/models/object_detection/retina_net/retina_net.py:221]: Yeah I will Add tests!

Comment (IMvision12) [reply keras_cv/models/object_detection/retina_net/retina_net.py:221]: Where exactly do I need to add the test?

Comment (bhack) [reply keras_cv/models/object_detection/retina_net/retina_net.py:221]: Yes like this:
https://github.com/keras-team/keras-cv/blob/d7fe322699275c965fc25e63bc0fa70c50f1b225/keras_cv/models/object_detection/retina_net/retina_net_test.py#L57

File: keras_cv/models/object_detection/retina_net/retina_net.py [hunk 2]
```diff
@@ -218,6 +218,11 @@ def _forward(self, images, training=None):
...
+            raise ValueError(
+                "`RetinaNet` does not yet support `Ragged Tensor Input`"
```
Comment (LukeWood) [inline keras_cv/models/object_detection/retina_net/retina_net.py:223]: "`RetinaNet()` does not yet support inputs of type `RaggedTensor` for input images. "
"To correctly resize your images for object detection tasks, we recommend resizing using "
"`keras_cv.layers.Resizing(pad_to_aspect_ratio=True, bounding_box_format=your_format)` "
"on your inputs."

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
Left a re-written error message, update and then LGTM!

File: keras_cv/models/object_detection/retina_net/retina_net.py (hunk 2 above)
Comment (IMvision12) [reply keras_cv/models/object_detection/retina_net/retina_net.py]: Done !

**Comment by IMvision12** [comment]
@LukeWood do i need to add tests for this ?

**Comment by IMvision12** [comment]
@LukeWood any more changes?

**Review by LukeWood — APPROVED** [review]
LGTM, running GCBRun

**Review by ianstenbit — APPROVED** [review]

**Pull request merged**